framework/src/araui/utils/emoji.c

```diff
@@ -40,39 +47,16 @@ static const uint8_t *g_emoji_assets[UTF8_MAX_EMOJI_COUNT] = {
 	__emoji_u1F641, __emoji_u1F642, __emoji_u1F643, __emoji_u1F644
 };
 #else
-static const uint8_t *g_emoji_assets[UTF8_MAX_EMOJI_COUNT] = { NULL, };
+static const uint8_t *g_emoji_bitmap[UTF8_MAX_EMOJI_COUNT] = { NULL, };
 #endif
 
-static ui_asset_t g_emoji[UTF8_MAX_EMOJI_COUNT];
-
-void emoji_init(void)
-{
-	int i;
-	for (i = 0; i < UTF8_MAX_EMOJI_COUNT; i++) {
-		if (g_emoji_assets[i]) {
-			g_emoji[i] = ui_image_asset_create_from_buffer(g_emoji_assets[i]);
-		}
-	}
-}
-
-void emoji_deinit(void)
-{
-	int i;
-	for (i = 0; i < UTF8_MAX_EMOJI_COUNT; i++) {
-		if (g_emoji[i]) {
-			ui_image_asset_destroy(g_emoji[i]);
-			g_emoji[i] = UI_NULL;
-		}
-	}
-}
-
-ui_asset_t emoji_get_asset(uint32_t utf_code)
+ui_bitmap_data_t *emoji_get_bitmap(uint32_t utf_code)
```

Comment:
'is_emoji' should check whole boundary of emoji including all list in unicode.org
Then emoji_get_bit map should return bitmap data if you already defined it, otherwise, default image or NULL.
Please assume that mms data from phone which including emoticon value is greater than 1f644, maybe face of animal as I remember.

Reply:
You're right. But currently, not supported character is shown as a blank. There is no fault issue.
Also.. full feature of emoji is not an urgent requestment.